pkg/webhooks/policymutation.go

```diff
@@ -287,14 +290,12 @@ func generateRuleForControllers(rule kyverno.Rule, controllers string, log logr.
 	}
 
 	// scenario A
-	if controllers == "all" {
-		if match.ResourceDescription.Name != "" || match.ResourceDescription.Selector != nil ||
-			exclude.ResourceDescription.Name != "" || exclude.ResourceDescription.Selector != nil {
-			log.Info("skip generating rule on pod controllers: Name / Selector in resource decription may not be applicable.", "rule", rule.Name)
-			return kyvernoRule{}
-		}
-		controllers = engine.PodControllers
+	if match.ResourceDescription.Name != "" || match.ResourceDescription.Selector != nil ||
```

Comment:
Please read the design doc to understand the logic: https://github.com/nirmata/kyverno/wiki/Auto-generating-rule-for-pod-controllers